src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  const stateCopy = {};
+
+  Object.assign(stateCopy, state);
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      for (const key in action.extraData) {
+        stateCopy[key] = action.extraData[key];
+      }
+
+      stateHistory.push(stateCopy);
```

Comment:
Pushing `stateCopy` directly to `stateHistory` will result in all entries in `stateHistory` being the same object. Consider creating a new copy of `stateCopy` before pushing it to `stateHistory`.